lib/nmatrix/math.rb

```diff
@@ -364,6 +364,37 @@ def cov(opts={})
     deviation_scores.transpose.dot(deviation_scores) / denominator
   end
 
+  # Raise a square matrix to a power. Be careful of buffer overflows!
+  # In case *n* is 0, a matrix of ones of the same dimension is returned. In case
+  # of negative *n*, the matrix is inverted and the absolute value of *n* taken 
+  # for computing the power.
+  # 
+  # == Arguments
+  # 
+  # * +n+ - Integer to which self is to be raised.
+  def pow n
+    raise ShapeError, "Only works with 2D square matrices." if shape[0] != shape[1]
+    raise TypeError, "Only works with integer powers" unless n.is_a?(Integer)
+
+    product = NMatrix.diag [1]*shape[0] 
```

Comment:
What do I do if I want to upcast a whole bunch of dtypes to `int64`?

Something like `NMatrix.upcast([:byte, :int8,:int16,:int32], :int64)`. Is this possible?